service/src/java/org/apache/hive/service/cli/thrift/RetryingThriftCLIServiceClient.java

```diff
@@ -310,7 +310,7 @@ protected synchronized TTransport connect(HiveConf conf) throws HiveSQLException
 
     String host = conf.getVar(HiveConf.ConfVars.HIVE_SERVER2_THRIFT_BIND_HOST);
     int port = conf.getIntVar(HiveConf.ConfVars.HIVE_SERVER2_THRIFT_PORT);
-    int maxThriftMessageSize = (int) conf.getSizeVar(HiveConf.ConfVars.HIVE_THRIFT_CLIENT_MAX_MESSAGE_SIZE);
+    int maxThriftMessageSize = (int) Math.min(conf.getSizeVar(HiveConf.ConfVars.HIVE_THRIFT_CLIENT_MAX_MESSAGE_SIZE),Integer.MAX_VALUE);
```

Comment:
Maybe add a new method to HiveConf like getSizeVar that takes in a long min, and a long max.
(It may even make sense to have it LOG.info when the gets clamped (orig val > max, orig val < min), but I am not sure if that would make logs too chatty).
Something like:
getSizeVarWithRange(HiveConf.ConfVars.HIVE_THRIFT_CLIENT_MAX_MESSAGE_SIZE, Integer.MIN_VALUE, Integer.MAX_VALUE). I'm bad at naming methods, so pick something that makes sense (though maybe WithRange does?).

Reply:
@jfsii I am not sure if we need to wrap with Integer.MIN_VALUE in this case. As we have a min value -1  check in sizeValidator and we set maxThriftMessageSize only if > 0 otherwise accepts thrift default (100 MB)

Reply:
I'm not sure the sizeValidator gets executed when a hive-site.xml is loaded (only when a SET is called), I'm not 100% sure though. That is the reason for my paranoia. Also having a helper method for that does the min and max range would make it usable in other instances in which we need to convert a "size" to an int or some other range.
It is not an absolute requirement, mostly a suggestion - if @saihemanth-cloudera is okay with the state of the patch that is good enough with me. (Functionally it is correct which is what matters).

Reply:
I'm confident about sizeValidator being executed when hive conf is loaded. The fact that other configs are also using sizeValidator confirms my assumption. So, I'm +1 on the patch.